modules/middleware/src/aggregations.js

```diff
@@ -174,14 +175,14 @@ export default class AggregationProcessor {
               // There is a query but it is just the nested field - just add it to aggs
               Object.assign(nested_aggs, nested_field_agg);
             }
+          } else {
+            const filteredAgg = this.create_filtered_agg(
+              field,
+              query,
+              nested_field_agg,
+            );
+            Object.assign(nested_aggs, filteredAgg);
           }
-
-          const filteredAgg = this.create_filtered_agg(
-            field,
-            query,
-            nested_field_agg,
-          );
-          Object.assign(nested_aggs, filteredAgg);
```

Comment:
This was not inside an else case before, looks like an initial porting error so I matched it up based on the original python code